### PR TITLE
Add Vector Projection Expression

### PR DESCRIPTION
**Pull request by UnderscoreTud**
### Description
This PR adds an expression to get the vector projection of two vectors.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/4725

### CODE REVIEW

**Comment by TheLimeGlass** [comment]
I feel like a better example should be in place. Otherwise no one is going to find use in this.

**Review by AyhamAl-Ali — APPROVED** [review]
LGTM ⚡

File: src/main/java/ch/njol/skript/expressions/ExprVectorProjection.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Vectors - Vector Projection")
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprVectorProjection.java:35]:
```suggestion
@Name("Vector Projection")
```

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprVectorProjection.java (hunk 1 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprVectorProjection.java:35]: Let's keep that part, see docs
image

**Pull request merged**